### PR TITLE
Feature test for User can login successfully if right info is inserted

### PR DESCRIPTION
**Pull request by LauraRe**
*As a user*
*In order to access my account*
*I should be able to login*

### Happy path : 
Test that user is able to login.
### Sad path : 
Test that error message is issued if user inserts wrong email or password.

### CODE REVIEW

**Review by Joberg95 — APPROVED** [review]
Good job!

**Review by aleciabenjamin — CHANGES_REQUESTED** [review]
no newline at end of page

**Review by aleciabenjamin — APPROVED** [review]

**Review by perhegg — APPROVED** [review]
Nice!

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -20,6 +20,7 @@ def conversation
   protected
 
   def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
```
Comment (faraznaeem) [inline app/controllers/application_controller.rb]: Please explain what this line of code does, and why you are fixing this as this code has to do with creating a new user in the system and you are working on the login of a user?

File: features/login.feature [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Background:
+    Given I visit the "landing" page
+    And The following user exist
+        | name     | email                | password    | password_confirmation |
+        | Per Hegg | per_hegg@hotmail.com | pereriksson | pereriksson           |
+
+
+Scenario: User can successfully login [Happy Path]
+
```
Comment (faraznaeem) [inline features/login.feature]: remove extra line

File: features/login.feature [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    When I click on "Login"    
+    And I fill in "Email" with "per_hegg@hotmail.com"
+    And I fill in "Password" with "pereriksson"
+    And I click on "Log in"
+    Then I should see "Signed in successfully."
+
+Scenario: User can't login [Sad Path]
+    Given I click on "Login"
+        When I fill in "Email" with " "
```
Comment (faraznaeem) [inline features/login.feature]: mind the indentation

File: features/login.feature [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Scenario: User can successfully login [Happy Path]
+
+    When I click on "Login"    
+    And I fill in "Email" with "per_hegg@hotmail.com"
+    And I fill in "Password" with "pereriksson"
+    And I click on "Log in"
+    Then I should see "Signed in successfully."
+
+Scenario: User can't login [Sad Path]
+    Given I click on "Login"
```
Comment (faraznaeem) [inline features/login.feature]: Please use the same flow for all of your scenarios

File: features/step_definitions/basic_steps.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+Given("I visit the {string} page") do |string|
+    visit '/'
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: use the rails path helper methods here, please look at the extras in the course material for more guidance

File: features/step_definitions/basic_steps.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+end
+
+Given("The following user exist") do |table|
+    table.hashes.each do |user|
+        User.create!(user)
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: Use factory here instead

File: features/step_definitions/basic_steps.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    end
+end
+
+When("I click on {string}") do |string|
+    click_on(string)
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: don't use `string` as a variable name, and be consistent with the with the brackets

File: features/step_definitions/basic_steps.rb [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+end
+
+When("I fill in {string} with {string}") do |string, string2|
+    fill_in string, with: string2
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: choose better variable names

**Review by faraznaeem — CHANGES_REQUESTED** [review]

**Comment by faraznaeem** [comment]
Update title and add a user story if you have one to the description

**Review by Joberg95 — APPROVED** [review]
looks good now, good job

**Comment by faraznaeem** [comment]
Better PR title in the future, but good job

**Review by aleciabenjamin — COMMENTED** [review]
lgtm

**Pull request merged**